### ISSUE TITLE
[Bug]: A fatal error has been detected by the Java Runtime Environment SIGSEGV (0xb)

### ISSUE THREAD

**Issue by jebbench** [adoptium/containers#336]
### Please add the exact image (with tag) that you are using

eclipse-temurin:jdk-17

### Please add the version of Docker you are running

Google Kubernetes Engine

### What happened?

JVM crashed (crash log below) when under light load.

Unfortunately it was running on a container which was restarted after the crash so I can't access the error report files.

We've seen the crash occur twice, both times when the application was under light load.

~~The image was built using JIB (https://github.com/GoogleContainerTools/jib).~~ We've now seen this occur in a container that was not built using JIB.

Heap usage was around 1gb of 2gb allocated.

### Relevant log output

```shell
[error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007f3ea639d898]
#
# https://github.com/adoptium/adoptium-support/issues
# If you would like to submit a bug report, please visit:
#
# //replay_pid1.log
# Compiler replay data is saved as:
#
# //hs_err_pid1.log
# An error report file with more information is saved as:
#
# JFR recording file will be written. Location: //hs_err_pid1.jfr
#
# Core dump will be written. Default location: /core.%e.1.%t
#
# C 0x0000000000000000
# Problematic frame:
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (17.0.5+8, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# JRE version: OpenJDK Runtime Environment Temurin-17.0.5+8 (17.0.5+8) (build 17.0.5+8)
#
# SIGSEGV (0xb) at pc=0x0000000000000000, pid=1, tid=19
#
# A fatal error has been detected by the Java Runtime Environment:
#
```

**Comment by karianna** [comment]
@jebbench We'll need the crash log to investigate.  On the off chance this is fixed in 17.0.6 then you can wait a couple of weeks for that release.

**Comment by jebbench** [comment]
@karianna this is still occuring using 17.0.6.

I will see if there's a way to retrieve the crash logs.

**Comment by jebbench** [comment]
@karianna I've been able to get the crash log:
[hs_err_pid1.log](https://github.com/adoptium/containers/files/10708410/hs_err_pid1.log)

Let me know if there's anything else we can do or provide to help.

**Comment by karianna** [comment]
Ah fn this is a Kotlin sourced crash, what version of Kotlin are you running?

**Comment by jebbench** [comment]
Afraid so, the application was built using Kotlin `1.7.21`.

**Comment by karianna** [comment]
> Afraid so, the application was built using Kotlin `1.7.21`.

Any chance you're able to move to 1.8.10?

**Comment by jebbench** [comment]
Yes - we should be able to migrate to 1.8.10, but it could be a few weeks before I can get another crash log unless we get lucky and see one in our test environment (unless the upgrade fixes it 🤞).

I'll start migrating our projects.

**Comment by jebbench** [comment]
Crash report using Kotlin 1.8.10.
[hs_err_pid1.log](https://github.com/adoptium/containers/files/10831575/hs_err_pid1.log)

**Comment by karianna** [comment]
Interesting, also was reported https://github.com/adoptium/adoptium-support/issues/659

**Comment by karianna** [comment]
@jebbench Apologies for the run around but https://youtrack.jetbrains.com/issues/kt is the place you'll want to report this as Kotlin is emitting something that the C2 compiler doesn't like.

**Comment by jebbench** [comment]
Thanks @karianna, I have raised this as https://youtrack.jetbrains.com/issue/KT-56975/JVM-A-fatal-error-has-been-detected-by-the-Java-Runtime-Environment-SIGSEGV-0xb

**Comment by karianna** [comment]
Thanks, closing here (but if the Kotlin folks bounce it back we can look at opening up at OpenJDK proper itself).

**Comment by ilmirus** [comment]
@karianna

Hi, Kotlin Team here. Yes, it looks very much as a OpenJDK bug. The thing with coroutines is that we generate code, which is not possible to write in Java - we have non-canonical loops or jumps to catch blocks, which are likely to not be covered by tests. There was even an anekdot, that Graal could not compile coroutines code and run the whole module in interpreter mode. Additionally, if there was a bug in generated bytecode, it would lead to VerifyError. Also, there is an issue reported with same JDK version and same symptoms - SIGSEGV in C2 on coroutines code and is very sporadic https://youtrack.jetbrains.com/issue/KT-54693/SIGSEGV-0xb-at-pc0x0000000000000000-C2-CompilerThread0.

**Comment by karianna** [comment]
Thanks @ilmirus - I'll get this reported upstream!

**Comment by karianna** [comment]
https://bugs.openjdk.org/browse/JDK-8303279
